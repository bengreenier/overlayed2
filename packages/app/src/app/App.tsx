import React from 'react'
import { Sand } from '../cw_shape/Sand'
import { Box, Grommet } from 'grommet'
import { useAuth } from './context/Auth'
import AuthenticatedApp from './authenticated/AuthenticatedApp'
import UnauthenticatedApp from './unauthenticated/UnauthenticatedApp'
import { theme } from './util/theme'

/**
 * The top level app experience
 *
 * Chooses between authenticated and unauthenticated experiences
 */
const App: React.FC = () => {
  // we need to know if the user is authenticated, so we consume the auth context
  const { loading, isAuthenticated } = useAuth()

  // if we don't yet know about auth state, show a loading indicator
  if (loading) {
    // note: this fades in after 1000ms, so if loading takes less than that nothing shows
    // this prevents the ugly "single frame loader" visual
    return (
      <Box width="medium" align="center" animation={{ delay: 1000, type: 'fadeIn' }}>
        <Sand />
      </Box>
    )
  }

  // we wrap our whole app in a grommet context so that theming and formatting for
  // grommet works correctly
  return (
    <Grommet theme={theme}>
      {isAuthenticated ? <AuthenticatedApp /> : <UnauthenticatedApp />}
    </Grommet>
  )
}

export default App
