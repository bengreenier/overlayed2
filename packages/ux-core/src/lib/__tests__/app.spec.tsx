import React from 'react'
import App from '../app'
import renderer from 'react-test-renderer'

describe('app', () => {
  it('should render', () => {
    const component = renderer.create(<App />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
