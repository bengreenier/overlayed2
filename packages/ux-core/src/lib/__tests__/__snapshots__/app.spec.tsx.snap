// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render 1`] = `
<div>
  <p>
    overlayed2 ux-core app
  </p>
</div>
`;
